rails-commenting.rb

```diff
@@ -6,39 +6,39 @@
 
 # FILE: app/controller/blog_posts_controller.rb
 
-# ---1)
+# ---1) Line 10 starts off the blog_posts_controller.rb page and is automatically generated when ($ rails g controller BlogPost) is ran. Next, a method for index is created using def (endwise extension makes sure that 'end's are automatically generated upon method creation).
 class BlogPostsController < ApplicationController
   def index
-    # ---2)
+    # ---2) Calling .all on BlogPost will allow us to see every instance of the BlogPost class, and then THAT is saved into an instance variable (@posts). Calling that instance variable from the index method lets us see all instances of BlogPost.
     @posts = BlogPost.all
   end
 
-  # ---3)
+  # ---3) We call .find on BlogPost and specify that we're looking for a parameter (params), and that that parameter should be 'id'. This allows us to find any instance of BlogPost that we want to if we put in a param of id. We save that into an instance variable (@post). This all happens in a new method called show. That lets us put down an id param to the URL, and find the specific instance of BlogPost we want. We can put down on the show.html.erb page that we simply, want the application to show us that instance of BlogPost.
   def show
     @post = BlogPost.find(params[:id])
   end
 
-  # ---4)
+  # ---4) We call .new on BlogPost and save it to an instance variable (@post) inside a newly created method called 'new'. This allows us to make a new instance of the class BlogPost with one command from the 'new' method if we call the @post instance variable. Line 26 starts a new method called 'create'.
   def new
     @post = BlogPost.new
   end
 
   def create
-    # ---5)
+    # ---5) We call .create on BlogPost with the argument of (blog_post_params) and save it to an instance variable (@post). I should note right about now that although we have multiple instance variables of @post, they're all created under different methods and are each locally scoped, so there is no problem with them overriding each other. But what this does is, allows an instance of BlogPost to be created inside the path of blog_post_params. Then we say that if this locally scoped instance variable @post is valid, then we want to redirect ourselves to blog_post_path which accepts an argument of instance variable @post. On line 34 we start creating a new method called 'edit'.
```

Comment:
Instance variables are not locally scoped but all variables are time scope and so when another method is activated the new variable takes over.